Improve info log messages

tgraf opened the pull request:
Auditing all info messages as per #7201. First batch of changes. PR is split into small individual changes for simplified review.

Related: #6776
Related: #6989
Related: #7201

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/7198)
<!-- Reviewable:end -->

tgraf committed:
bpf: Provide warning message when BPF filesystem is not mounted

It was not possible to determine based on log files when the BPF filesystem was
required to be mounted or not.

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
ctmap: Remove unnecessary garbage collection info message

The metrics are good enough at this point that the following regular info
message is no longer needed:
```
Feb 19 11:17:00 k8s1 cilium-agent[2833]: level=info msg="Conntrack garbage collection statistics" completed=true duration="401.513µs" family=ipv4 maxEntries=262144 numDeleted=0 numKeyFallbacks=0 numLookups=1 numLookupsFailed=0 protocol=non-TCP startTime="2019-02-19 11:17:00.017800041 +0000 UTC m=+4142.655480343" subsys=map-ct
```

Related: #6989

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
agent: Improve log messages when cleaning stale endpoint interfaces

Use log.Warning instead of printing warnings to stdout

Related: #6776

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
k8s: Print apiserver address in info message when connecting

Related: #6776

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
k8s: Remove redundant info message when deriving nodeIP & PodCIDR

Related: #6776

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
k8s: Remove duplicate namespace info message

Any non-standard namespace setting is printed at the beginning:
```
Feb 22 20:25:58 k8s1 cilium-agent[13012]: level=info msg="  --k8s-namespace=''" subsys=daemon
```

No need to print it again.

Related: #6776

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
ipam: Print address ranges in info message

Related: #6776

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
k8s: Print addresses used to annotate the node

Related: #6776

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
ipam: Print node addresses even if address family is disabled

IPv4 node addressing can be relevant even if IPv4 is disabled, for example when
used for encapsulation addressing.

Related: #6776

Signed-off-by: Thomas Graf <thomas@cilium.io>

ianvernon commented on pkg/bpf/bpffs_linux.go:
```diff
@@ -130,6 +131,16 @@ var (
 // mountFS mounts the BPFFS filesystem into the desired mapRoot directory.
 func mountFS() error {
+	if k8s.IsEnabled() {
+		log.Warning("================================= WARNING ==========================================")
```
Nice, this will hopefully reduce confusion for people who aren't mounting it correctly.

ianvernon commented on pkg/bpf/bpffs_linux.go:
```diff
@@ -185,6 +196,8 @@ func checkOrMountCustomLocation(bpfRoot string) error {
 			return err
 		}
+
+		return nil
```
Why return nil here now vs. before?

tgraf replied:
It's to ensure that we print `log.Infof("Detected mounted BPF filesystem at %s", mapRoot)` only when the BPF FS was not mounted by the agent itself.

tgraf replied:
Happy to restructure this if you have a better idea. The main motivation is to be able to grep for a particular message to proof that the BPF filesystem was already mounted.

ianvernon replied:
@tgraf makes sense, thanks!

tgraf committed:
health: Warn on failures to cleanup health endpoint

Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
health: Convert route add info message to debug

Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
health: Remove unused logFromCommand() function

Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
agent: More accurate info message when restore is disabled

Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
k8s: Convert service definition changed message from info to debug

Related: #6989
Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
health: Remove redundant 'Building health endpoint' message

Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
agent: Provide summary of service ID restore

Printing an info message for each restore service ID can be incredibly verbose
in environments with many services. Turn the existing message into a debug and
provide a summary.

Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
agent: Improve logs when restoring services from BPF maps

Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
agent: Improve log messages on restore

Use warnings when endpoints cannot be restored for some reason and include the
Kubernetes pod name for easier identification.

Ignore the health endpoint when providing restore statistics to simplify
assessment of whether restore was successful or not.

Related: #6776
Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
bpf: Use warnings for map metadata mismatch

Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
conntrack: Use debug message to indicate that GC is skipped for ENOENT

Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
ctmap: Warn if CT map was upgraded as it results in disruption

Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
k8s: Remove info message when parsing headless services

Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
logging: Remove unused setFireLevels() function

Related: #7201

Signed-off-by: Thomas Graf <thomas@cilium.io>

aanm commented on daemon/daemon.go:
```diff
@@ -1085,7 +1085,10 @@ func NewDaemon(dp datapath.Datapath) (*Daemon, *endpointRestoreState, error) {
-	log.Info("Initializing IPAM")
+	log.WithFields(logrus.Fields{
+		logfields.V4Prefix: dp.LocalNodeAddressing().IPv4().AllocationCIDR(),
+		logfields.V6Prefix: dp.LocalNodeAddressing().IPv6().AllocationCIDR(),
```
Won't this panic if `enable-ipv6: false`?

ianvernon replied:
Good catch @aanm, +1.  am weary of doing a lot of `a.b.c.d` in the code because it has bitten us in the past in cases where `c` is nil.

tgraf replied:
I was worried about this and tested this. It returns "nil"

ianvernon commented on daemon/state.go:
```diff
@@ -89,13 +90,18 @@ func (d *Daemon) restoreOldEndpoints(dir string, clean bool) (*endpointRestoreSt
-		scopedLog := log.WithField(logfields.EndpointID, ep.ID)
+		scopedLog := log.WithFields(logrus.Fields{
+			logfields.EndpointID: ep.ID,
+			"k8sPodName":         ep.GetK8sNamespaceAndPodNameLocked(),
```
Nit - only add this `k8sPodName` field @if Kubernetes configuration is enabled.
